test/integration/install/test_install.py

```diff
@@ -645,3 +645,101 @@ def test_install_tt_fail_exit_code_dependency_check(
     )
     install_process_rc = install_process.wait()
     assert install_process_rc == 1
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("old_tarantool_version, expected_install_msg", [
+    pytest.param("7192bf6", "Found newest commit of tarantool in master"),
+    pytest.param("master", "tarantool_master version of tarantool already exists")
+])
+def test_install_tarantool_fetch_latest_version(
+        tt_cmd,
+        tmp_path,
+        old_tarantool_version,
+        expected_install_msg):
+    config_path = os.path.join(tmp_path, config_name)
+    # Create test config.
+    with open(config_path, "w") as f:
+        yaml.dump({"env": {"bin_dir": "", "inc_dir": "./my_inc"}}, f)
+
+    tmp_path_without_config = tempfile.mkdtemp()
+
+    # Install not latest version of tarantool by commit hash.
+    install_cmd = [tt_cmd, "--cfg", config_path, "install",
+                   "-f", "tarantool", old_tarantool_version]
+    instance_process = subprocess.Popen(
+        install_cmd,
+        cwd=tmp_path_without_config,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.DEVNULL,
+        text=True
+    )
+    instance_process_rc = instance_process.wait()
+    assert instance_process_rc == 0
+
+    # Renaming installed 'old' tarantool to 'master'
+    # and change symlink to old version to simulate
+    # that this is the latest version.
+    os.rename(os.path.join(tmp_path, "bin", "tarantool_" +
+              old_tarantool_version), os.path.join(tmp_path, "bin", "tarantool_master"))
+    os.remove(os.path.join(tmp_path, "bin", "tarantool"))
+    os.symlink(os.path.join(tmp_path, "bin", "tarantool_master"),
+               os.path.join(tmp_path, "bin", "tarantool"))
+    os.remove(os.path.join(tmp_path, "my_inc", "include", "tarantool"))
+    os.rename(os.path.join(tmp_path, "my_inc", "include",
+              "tarantool_" + old_tarantool_version),
+              os.path.join(tmp_path, "my_inc", "include", "tarantool_master"))
+
+    # Installing newest version.
+    install_cmd = [tt_cmd, "--cfg", config_path, "install",
+                   "-f", "tarantool", "master"]
+    instance_process = subprocess.Popen(
+        install_cmd,
+        cwd=tmp_path_without_config,
+        stdin=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        text=True
+    )
+    instance_process.stdin.write("y\n")
+    instance_process.stdin.close()
+
+    instance_process_rc = instance_process.wait()
+    assert instance_process_rc == 0
+
+    output = instance_process.stdout.read()
+    assert expected_install_msg in output
+
+
+@pytest.mark.slow
+def test_install_tt_fetch_latest_version(tt_cmd, tmp_path):
```

Comment:
Please add the following test:
bin/
    tt -> tt_2.3.1
    tt_2.3.1
    tt_master

Newer commit found, or not.

Reply:
Added.